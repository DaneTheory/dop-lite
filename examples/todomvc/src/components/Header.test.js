import React from 'react'
import renderer from 'react-test-renderer';
import Header from './Header'


test('Adding a new todo', () => {
  const component = renderer.create(
    <Header />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually changed input new todo
  let newtodotext = "New todo"
  tree.children[1].props.onChange({target:{value:newtodotext}})
  // re-rendering
  tree = component.toJSON();
  expect( tree.children[1].props.value).toBe(newtodotext)

  // saving new todo manually
  tree.children[1].props.onKeyDown({which:13,target:{value:newtodotext}})
  // re-rendering
  tree = component.toJSON();
  expect(tree.children[1].props.value).toBe("")
  expect(tree).toMatchSnapshot();
});

