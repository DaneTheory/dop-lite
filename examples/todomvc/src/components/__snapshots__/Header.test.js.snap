// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Adding a new todo 1`] = `
<header
  className="header"
>
  <h1>
    todos
  </h1>
  <input
    autoFocus="true"
    className="new-todo"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="What needs to be done?"
    type="text"
    value=""
  />
</header>
`;

exports[`Adding a new todo 2`] = `
<header
  className="header"
>
  <h1>
    todos
  </h1>
  <input
    autoFocus="true"
    className="new-todo"
    onChange={[Function]}
    onKeyDown={[Function]}
    placeholder="What needs to be done?"
    type="text"
    value=""
  />
  <input
    checked={false}
    className="toggle-al"
    onChange={[Function]}
    type="checkbox"
  />
</header>
`;
